tests/probe_identity.rs

```diff
@@ -0,0 +1,43 @@
+#[cfg(test)]
+mod tests {
+    use bmputil::probe_identity::ProbeIdentity;
+    #[test]
+    fn extract_native(){
+        let res: ProbeIdentity = "Black Magic Probe v2.0.0-rc2".to_string().into();
+
+        assert_eq!(res.probe_name(), Some("native".into()));
+        assert_eq!(res.version(), Some("v2.0.0-rc2".into()));
+    }
+
+    #[test]
+    fn extract_old(){
+        let res: ProbeIdentity = "Black Magic Probe".to_string().into();
+
+        assert_eq!(res.probe_name(), Some("native".into()));
+        assert_eq!(res.version(), Some("v1.6".into()));
+    }
+
+    #[test]
+    fn extract_st_link(){
+        let res: ProbeIdentity = "Black Magic Probe (ST-Link v2) v1.10.0-1273-g2b1ce9aee".to_string().into();
+
+        assert_eq!(res.probe_name(), Some("st-link v2".into()));
+        assert_eq!(res.version(), Some("v1.10.0-1273-g2b1ce9aee".into()));
+    }
+
+    #[test]
+    fn extract_without_closing(){
+        let res: ProbeIdentity = "Black Magic Probe (ST-Link".to_string().into();
+
+        assert_eq!(res.probe_name(), None);
+        assert_eq!(res.version(), None);
+    }
+
+    #[test]
+    fn unknown(){
+        let res: ProbeIdentity = "Something (v1.2.3)".to_string().into();
```

Comment:
One example that could be used here would be a Git commit hash (`g221c3031f` for example) which can happen if someone shallow clones the repo with depth 1 and so doesn't get the tags, or necessary history to link the commit up to the most recent tag.

We will also need to handle `Black Magic Probe v2.0.0-rc2-65-g221c3031f` for example, so it would be useful to include that as another specific test here due to some changes that will come in on the feature branch we mentioned before which introduces a VersionString type to handle the version strings for equality checking and some other functionality.

Reply:
I have used the example in the test, and that also seems to pass!